README.md

```diff
@@ -113,10 +125,27 @@ type SeatChangedEvent = JSONEventType<
 
 type ReservationEvents = SeatReservedEvent | SeatChangedEvent;
 
-async function simpleTest(): Promise<void> {
-  const streamName = "booking-abc123";
+const client = EventStoreDBClient.connectionString`esdb://localhost:2113?tls=false`;
+
+(async () => {
+  try {
+    const streamName = "booking-abc123";
+    console.log("Creating and appending events...");
+
+    await createAndAppendEvents(client, streamName);
+    console.log("Reading and processing events...");
 
-  const event = jsonEvent<SeatReservedEvent>({
+    const reservation = await readAndProcessEvents(client, streamName);
```

Comment:
If I was coming to a repo, I'd rather read a concise example, that can be read as a one shot, top to bottom, without needing to look at what's inside a function.
 
Possibly something like this closer rides the line of readable in a glance, with explanation of what you're looking at (via comments) and is still copy / paste runnable:

```ts
import {
  EventStoreDBClient,
  jsonEvent,
  FORWARDS,
  START,
  type JSONEventType,
} from "@eventstore/db-client";

// Create a client to connect to EventStoreDB
const client = EventStoreDBClient.connectionString`esdb://localhost:2113`;

// Define the events that can occur during a booking

type SeatReservedEvent = JSONEventType<
  "seat-reserved",
  {
    reservationId: string;
    movieId: string;
    userId: string;
    seatId: string;
  }
>;

type SeatChangedEvent = JSONEventType<
  "seat-changed",
  {
    reservationId: string;
    newSeatId: string;
  }
>;

type ReservationEvents = SeatReservedEvent | SeatChangedEvent;

// Write events to a stream that describe the history of our booking

const streamName = "booking-abc123";

const event = jsonEvent<SeatReservedEvent>({
  type: "seat-reserved",
  data: {
    reservationId: "abc123",
    movieId: "tt0368226",
    userId: "nm0802995",
    seatId: "4b",
  },
});

const appendResult = await client.appendToStream<ReservationEvents>(
  streamName,
  event
);

// By reading the events in the stream, we can construct the current state of the booking

interface Reservation {
  reservationId: string;
  movieId: string;
  userId: string;
  seatId: string;
}

const events = client.readStream<ReservationEvents>(streamName, {
  fromRevision: START,
  direction: FORWARDS,
  maxCount: 10,
});

const reservation: Partial<Reservation> = {};

for await (const { event } of events) {
  switch (event?.type) {
    case "seat-reserved": {
      reservation.reservationId = event.data.reservationId;
      reservation.movieId = event.data.movieId;
      reservation.seatId = event.data.seatId;
      reservation.userId = event.data.userId;
      break;
    }
    case "seat-changed": {
      reservation.seatId = event.data.newSeatId;
      break;
    }
  }
}

// Do something with our reservation
console.log(reservation);
```

Reply:
done 👍